zio-http/src/test/scala/zhttp/http/URLSpec.scala

```diff
@@ -4,7 +4,29 @@ import zio.test.Assertion._
 import zio.test._
 
 object URLSpec extends DefaultRunnableSpec {
-  def spec = suite("URL")(suite("fromString")(test("Should Handle invalid url String with restricted chars") {
-    assert(URL.fromString("http://mw1.google.com/$[level]/r$[y]_c$[x].jpg"))(isLeft)
-  }))
+
+  val fromString = suite("fromString")(
+    test("Should Handle invalid url String with restricted chars") {
+      assert(URL.fromString("http://mw1.google.com/$[level]/r$[y]_c$[x].jpg"))(isLeft)
+    },
+    test("Should Handle empty query string") {
+      assert(URL.fromString("http://yourdomain.com/list/users").map(_.queryParams))(
+        isRight(equalTo(Map.empty[String, List[String]])),
+      )
+    },
+    test("Should Handle query string") {
+      assert(
+        URL
+          .fromString("http://yourdomain.com/list/users?user_id=1&user_id=2&order=ASC&text=zio-http%20is%20awesome%21")
+          .map(_.queryParams),
+      )(
+        isRight(
+          equalTo(Map("user_id" -> List("1", "2"), "order" -> List("ASC"), "text" -> List("zio-http is awesome!"))),
+        ),
+      )
+    },
+  )
+
+  def spec =
+    suite("URL")(fromString)
```

Comment:
Any particular reason to create a variable? Can you inline it?
#nitpick

Reply:
Usually I have multiple tests around one method in a class, I favor splitting them out like this